mock/py/mockbuild/buildroot.py

```diff
@@ -576,11 +576,13 @@ def _lock_buildroot(self, exclusive):
         try:
             fcntl.lockf(self._lock_file.fileno(), lock_type | fcntl.LOCK_NB)
         except IOError:
+            self._lock_file = None
```

Comment:
What's the point of this reset?

Reply:
The idea is that the mock process deletes the lock file _only_ when it owns the lock. Otherwise, a mock process that wasn't able to lock the lockfile would still delete it in line 585 (see change below).

Reply:
Aha, can you please comment in-code?

Reply:
Sure!